di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/src/main/java/uk/gov/di/ipv/core/getcontraindicatorcredential/GetContraIndicatorCredentialHandler.java

```diff
@@ -96,96 +97,89 @@ public void handleRequest(InputStream input, OutputStream output, Context contex
+                                vcCi.setIssuanceDate(datebaseCi.getIssuanceDate().toString());
+                                vcCi.setMitigation(getMitigations(datebaseCi.getMitigations()));
+                            },
+                            () -> ciForVc.add(createContraIndicator(datebaseCi, txnUuid)));
+        }
+        return ciForVc;
     }
 
-    private List<Map<String, Object>> getContraIndicators(String userId) {
-        List<Map<String, Object>> contraIndicators = new ArrayList<>();
-        List<CimitStubItem> cimitStubItems = cimitStubItemService.getCIsForUserId(userId);
-        for (CimitStubItem cimitStubItem : cimitStubItems) {
-            Map<String, Object> contraIndicator = new LinkedHashMap<>();
-            contraIndicator.put(CODE, cimitStubItem.getContraIndicatorCode());
-            contraIndicator.put(ISSUERS, cimitStubItem.getIssuers());
-            contraIndicator.put(ISSUANCE_DATE, cimitStubItem.getIssuanceDate().toString());
-            contraIndicator.put(MITIGATION, getMitigations(cimitStubItem.getMitigations()));
-            contraIndicators.add(contraIndicator);
-        }
-        return contraIndicators;
+    private boolean ciShouldBeMerged(
+            CimitStubItem ciBeingProcessed, ContraIndicator ciAlreadyProcessed) {
+        boolean ciCodesMatch =
+                ciBeingProcessed.getContraIndicatorCode().equals(ciAlreadyProcessed.getCode());
+        boolean documentsMatch =
+                ciBeingProcessed.getDocument() == null
+                        ? ciAlreadyProcessed.getDocument() == null
+                        : ciBeingProcessed.getDocument().equals(ciAlreadyProcessed.getDocument());
+
+        return ciCodesMatch && documentsMatch;
     }
 
-    private List<Map<String, Object>> getMitigations(List<String> mitigationCodes) {
-        List<Map<String, Object>> mitigations = new ArrayList<>();
-        for (String mitigationCode : mitigationCodes) {
-            Map<String, Object> mitigation = new LinkedHashMap<>();
-            mitigation.put(CODE, mitigationCode);
-            mitigation.put(MITIGATION_CREDENTIAL, getMitigationCredentials());
-            mitigations.add(mitigation);
-        }
-        return mitigations;
+    private ContraIndicator createContraIndicator(CimitStubItem item, List<String> txnUuid) {
+        return new ContraIndicator(
+                item.getContraIndicatorCode(),
+                item.getDocument(),
+                item.getIssuanceDate().toString(),
+                new TreeSet<>(List.of(item.getIssuer())),
+                getMitigations(item.getMitigations()),
+                List.of(),
+                txnUuid);
     }
 
-    private List<Map<String, Object>> getMitigationCredentials() {
-        List<Map<String, Object>> mitigationCredentials = new ArrayList<>();
-        return mitigationCredentials;
+    private List<Mitigation> getMitigations(List<String> mitigationCodes) {
```

Comment:
Can `mitigationCodes` ever be `null`?

Reply:
No, if there are no mitigations on a CI we store it as an empty list in the DB. I suppose you could use the management API to manually set a CI with a null value for mitigations. Might be worth adding a guard clause here.